src/lib/markbind/src/parser.js

```diff
@@ -588,54 +576,53 @@ class Parser {
-      return element.map(el => this._rebaseReference(el, foundBase));
+      return element.map(el => this._rebaseReference(el));
     }
     if (Parser.isText(element)) {
       return element;
     }
-    // Rebase children element
-    const childrenBase = {};
+    // Rebase children elements
     element.children.forEach((el) => {
-      this._rebaseReference(el, childrenBase);
+      this._rebaseReference(el);
     });
-    // rebase current element
-    if (element.attribs[ATTRIB_INCLUDE_PATH]) {
-      const filePath = element.attribs[ATTRIB_INCLUDE_PATH];
-      let newBaseUrl = urlUtils.calculateNewBaseUrls(filePath, this.rootPath, this.baseUrlMap);
-      if (newBaseUrl) {
-        const { relative, parent } = newBaseUrl;
-        // eslint-disable-next-line no-param-reassign
-        foundBase[parent] = relative;
-      }
-      // override with parent's base
-      const combinedBases = { ...childrenBase, ...foundBase };
-      const bases = Object.keys(combinedBases);
-      if (bases.length !== 0) {
-        // need to rebase
-        newBaseUrl = combinedBases[bases[0]];
-        if (element.children) {
-          // ATTRIB_CWF is where the element was preprocessed
-          const currentBase = urlUtils.calculateNewBaseUrls(element.attribs[ATTRIB_CWF],
-                                                            this.rootPath, this.baseUrlMap);
-          if (currentBase && currentBase.relative !== newBaseUrl) {
-            cheerio.prototype.options.xmlMode = false;
-            const rendered = nunjucks.renderString(cheerio.html(element.children), {
-              // This is to prevent the nunjuck call from converting {{hostBaseUrl}} to an empty string
-              // and let the hostBaseUrl value be injected later.
-              hostBaseUrl: '{{hostBaseUrl}}',
-              baseUrl: `{{hostBaseUrl}}/${newBaseUrl}`,
-            }, { path: filePath });
-            element.children = cheerio.parseHTML(rendered, true);
-            cheerio.prototype.options.xmlMode = true;
-          }
-        }
-      }
-      delete element.attribs[ATTRIB_INCLUDE_PATH];
-    }
+
+    const includeSourceFile = element.attribs[ATTRIB_CWF];
+    const includedSourceFile = element.attribs[ATTRIB_INCLUDE_PATH];
     delete element.attribs[ATTRIB_CWF];
+    delete element.attribs[ATTRIB_INCLUDE_PATH];
+
+    // Skip rebasing for non-includes
+    if (!includedSourceFile || !element.children) {
+      return element;
+    }
+
+    // Where the children are included
+    const newBase = urlUtils.calculateNewBaseUrls(includedSourceFile, this.rootPath, this.baseUrlMap);
+    // Where the include was preprocessed
+    const currentBase = urlUtils.calculateImmediateParentSitePath(includeSourceFile,
+                                                                  this.rootPath, this.baseUrlMap);
+
+    // Only re-render if include src and content are from different sites
+    if (currentBase !== path.resolve(newBase.parent, newBase.relative)) {
```

Comment:
Then we can have currentBase !== newBase

Reply:
This is one of the use cases for `calculateNewBaseUrls` in particular
`relative` is needed for the resolving the new relative baseurl
`baseUrl: {{hostBaseUrl}}/${newBase.relative}`